src/APL/saleor-cloud/saleor-cloud-apl.ts

```diff
@@ -95,79 +103,172 @@ export class SaleorCloudAPL implements APL {
+          const message = `Cant parse response body: ${extractErrorMessage(e)}`;
 
-    if (!authData) {
-      debug("No auth data for given saleorApiUrl");
-      return undefined;
-    }
+          span.recordException(CloudAplError.RESPONSE_BODY_INVALID);
+          span
+            .setStatus({
+              code: SpanStatusCode.ERROR,
+              message,
+            })
+            .end();
 
-    return authData;
-  }
+          throw new SaleorCloudAplError(CloudAplError.RESPONSE_BODY_INVALID, message);
+        })) as CloudAPLAuthDataShape;
 
-  async set(authData: AuthData) {
-    debug("Saving data to SaleorCloudAPL for saleorApiUrl: %s", authData.saleorApiUrl);
+        const authData = authDataFromObject(mapAPIResponseToAuthData(parsedResponse));
+
+        if (!authData) {
+          debug("No auth data for given saleorApiUrl");
 
-    const response = await fetch(this.resourceUrl, {
-      method: "POST",
-      headers: { "Content-Type": "application/json", ...this.headers },
-      body: JSON.stringify(mapAuthDataToAPIBody(authData)),
-    }).catch((e) => {
-      debug("Failed to reach API call:  %s", extractErrorMessage(e));
-      debug("%O", e);
+          span.addEvent("Missing auth data for given saleorApiUrl");
+          span
+            .setStatus({
+              code: SpanStatusCode.OK,
+            })
+            .end();
 
-      throw new Error(`Error during saving the data: ${extractErrorMessage(e)}`);
-    });
+          return undefined;
+        }
+
+        span.setAttribute("appId", authData.appId);
+
+        span.end();
 
-    validateResponseStatus(response);
+        return authData;
+      }
+    );
+  }
 
-    debug("Set command finished successfully for saleorApiUrl: %", authData.saleorApiUrl);
+  async set(authData: AuthData) {
+    debug("Saving data to SaleorCloudAPL for saleorApiUrl: %s", authData.saleorApiUrl);
 
-    return undefined;
+    return this.tracer.startActiveSpan(
+      "Call SaleorCloudAPL SET",
+      {
+        attributes: {
+          saleorApiUrl: authData.saleorApiUrl,
+          appId: authData.appId,
+          [SemanticAttributes.PEER_SERVICE]: OTEL_APL_SERVICE_NAME,
+        },
+        kind: SpanKind.CLIENT,
+      },
+      async (span) => {
+        const response = await fetch(this.resourceUrl, {
+          method: "POST",
+          headers: { "Content-Type": "application/json", ...this.headers },
+          body: JSON.stringify(mapAuthDataToAPIBody(authData)),
+        }).catch((e) => {
+          debug("Failed to reach API call:  %s", extractErrorMessage(e));
+          debug("%O", e);
+
+          span.recordException(`Failed to reach API call:  ${extractErrorMessage(e)}`);
+          span
+            .setStatus({
+              code: SpanStatusCode.ERROR,
+            })
+            .end();
+
+          throw new Error(`Error during saving the data: ${extractErrorMessage(e)}`);
```

Comment:
Shouldn't `SaleorCloudAplError` be used here instead of the generic `Error` class?

Reply:
Good catch. Out of scope, but I updated this and few other instances (missed that some time ago clearly)